src/isar/state_machine/state_machine.py

```diff
@@ -264,33 +256,6 @@ def stop_mission(self):
         if not failure:
             self.mission_in_progress = False
 
-    def publish_step_status(self) -> None:
-        """Publishes the current step status to the MQTT Broker"""
-        payload: str = json.dumps(
-            {
-                "step_id": self.current_step.id if self.current_step else None,
-                "step_status": self.current_step.status if self.current_step else None,
-            },
-            cls=EnhancedJSONEncoder,
-        )
-
-        self.mqtt_client.publish(
-            topic=settings.ISAR_STEP_STATUS,
-            payload=payload,
-            retain=True,
-        )
-
-    def publish_mission(self) -> None:
-        payload: str = json.dumps(
-            {"mission": self.current_mission}, cls=EnhancedJSONEncoder
-        )
-
-        self.mqtt_client.publish(
-            topic=settings.ISAR_MISSION,
-            payload=payload,
-            retain=True,
-        )
```

Comment:
These have been removed but as far as I can tell not replaced. We do want to change them a little so we're not publishing the full mission every time etc. but we do need to publish these on change so the functionality should be there.

See my comment below on possibly needing event based publishing.